internal/app/go-filecoin/paymentchannel/manager.go

```diff
@@ -236,6 +218,105 @@ func PaychActorCtorExecParamsFor(client, miner address.Address) (initActor.ExecP
 	return p, nil
 }
 
+// GetMinerWorkerAddress mocks getting a miner worker address from the miner address
+func (pm *Manager) GetMinerWorkerAddress(ctx context.Context, miner address.Address, tok shared.TipSetToken) (address.Address, error) {
+	view, err := pm.stateViewer.GetStateView(ctx, tok)
+	if err != nil {
+		return address.Undef, err
+	}
+	_, fcworker, err := view.MinerControlAddresses(ctx, miner)
+	return fcworker, err
+}
+
+func (pm *Manager) WaitForCreatePaychMessage(ctx context.Context, mcid cid.Cid) (address.Address, error) {
+	var newPaychAddr address.Address
+
+	handleResult := func(b *block.Block, sm *types.SignedMessage, mr *vm.MessageReceipt) error {
+		var res initActor.ExecReturn
+		if err := encoding.Decode(mr.ReturnValue, &res); err != nil {
+			return err
+		}
+
+		var msgParams initActor.ExecParams
+		if err := encoding.Decode(sm.Message.Params, &msgParams); err != nil {
+			return err
+		}
+
+		var ctorParams *paychActor.ConstructorParams
+		if err := encoding.Decode(msgParams.ConstructorParams, &ctorParams); err != nil {
+			return err
+		}
+
+		newPaychAddr = res.RobustAddress
+		return nil
+	}
+
+	err := pm.waiter.Wait(pm.ctx, mcid, handleResult)
+	if err != nil {
+		return address.Undef, err
+	}
+	return newPaychAddr, nil
+}
+
+func (pm *Manager) AddFundsToChannel(paychAddr address.Address, amt abi.TokenAmount) (cid.Cid, error) {
+	var chinfo ChannelInfo
+	st := pm.paymentChannels.Get(paychAddr)
+	if err := st.Get(&chinfo); err != nil {
+		return cid.Undef, err
+	}
+
+	mcid, _, err := pm.sender.Send(context.TODO(), chinfo.From, paychAddr, amt, defaultGasPrice, defaultGasLimit, true, builtin.MethodSend, nil)
+	if err != nil {
+		return cid.Undef, err
+	}
+	// TODO: track amts in paych store
```

Comment:
This either needs to be completed this PR or tagged with a go-filecoin issue number.